src/net/gateway-discovery.ts

```diff
@@ -1,24 +1,16 @@
 import { ContentUri } from "../wrappers/content-uri"
-// import GatewayInfo from "../wrappers/gateway-info"
 import { Gateway } from "./gateway"
 import { IDVoteGateway } from "./gateway-dvote"
 import { IWeb3Gateway } from "./gateway-web3"
 import { EthNetworkID, GatewayBootnode } from "./gateway-bootnode"
-import { parseURL } from 'universal-parse-url'
 import { GATEWAY_SELECTION_TIMEOUT } from "../constants"
 import { JsonBootnodeData } from "../models/gateway"
-import { promiseFuncWithTimeout, promiseWithTimeout } from "../util/timeout"
+import { promiseFuncWithTimeout } from "../util/timeout"
 import { Random } from "../util/random"
 import { VocdoniEnvironment } from "../models/common"
+import { GatewayDiscoveryError, GatewayDiscoveryValidationError } from "../util/errors/gateway-discovery"
 
-
-const PARALLEL_GATEWAY_TESTS = 5
-
-// Minimum number of GW's that need to succeed on a check round for it to be valid.
-// Otherwise, the next threshold rate will be attempted
-const MIN_ROUND_SUCCESS_COUNT = 2
-
-export type IGatewayDiscoveryParameters = {
+export interface IGatewayDiscoveryParameters {
```

Comment:
@marcvelmer  FYI this is possibly breaking, we should add it on the changelog file